public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -27,123 +28,139 @@ const RECEIVER_STATUS_POLLING_INTERVAL = 10 * 1000; // 10 seconds
  * 3. (if available) additional metadata about Grafana Managed contact points
  * 4. (if available) the OnCall plugin metadata
  */
-interface UseContactPointsWithStatusOptions {
-  includePoliciesCount: boolean;
-  receiverStatusPollingInterval?: number;
-}
 
-const defaultHookOptions = {
-  includePoliciesCount: true,
-  receiverStatusPollingInterval: RECEIVER_STATUS_POLLING_INTERVAL,
+const {
+  useGetAlertmanagerConfigurationQuery,
+  useGetContactPointsListQuery,
+  useGetContactPointsStatusQuery,
+  useGrafanaNotifiersQuery,
+  useLazyGetAlertmanagerConfigurationQuery,
+  useUpdateAlertmanagerConfigurationMutation,
+} = alertmanagerApi;
+
+const { useGrafanaOnCallIntegrationsQuery } = onCallApi;
+
+/**
+ * Check if OnCall is installed, and fetch the list of integrations if so.
+ *
+ * Otherwise, returns no data
+ */
+const useOnCallIntegrations = ({ skip }: { skip?: boolean } = {}) => {
+  const { installed, loading } = usePluginBridge(SupportedPlugin.OnCall);
+  const oncallIntegrationsResponse = useGrafanaOnCallIntegrationsQuery(undefined, { skip: skip || !installed });
+
+  return useMemo(() => {
+    if (installed) {
+      return oncallIntegrationsResponse;
+    }
+    return {
+      isLoading: loading,
+      data: undefined,
+    };
+  }, [installed, loading, oncallIntegrationsResponse]);
+};
+
+/**
+ * Fetch contact points from separate endpoint (i.e. not the Alertmanager config) and combine with
+ * OnCall integrations and any additional metadata from list of notifiers
+ * (e.g. hydrate with additional names/descriptions)
+ */
+export const useGetContactPoints = () => {
```

Comment:
Split out so we can more easily slot in the k8s API into this method. As per the description, we were slightly conflating the logic with "also fetch me the policies". Splitting out makes it a bit clearer to reason about what the hook is used for